### ISSUE TITLE
Do not compare enum values with null

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#996]
We should report an issue in case an enum value is compared to `null`. As they are enum values, they are never `null` (except the enum type itself is a reference type and no value type).

Following should trigger issues:
```c#
public StringComparison Comparison { get; set; }

public void DoSomething()
{
    if (Comparison != null) { }
}
```
same for
```c#
public StringComparison Comparison { get; set; }

public void DoSomething()
{
    if (Comparison == null) { }
}
```
